### PR TITLE
TextField: Convert to flexbox, support addons

### PR DESCRIPTION
**Pull request by micahgodbolt**
#### Pull request checklist

- [X] Addresses an existing issue: #1383

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/TextField/TextField.tsx [hunk 1]
```diff
@@ -224,6 +234,13 @@ export class TextField extends BaseComponent<ITextFieldProps, ITextFieldState> i
     }
   }
 
+  private _onRenderAddon(props): JSX.Element {
+    let { addonString } = props;
+    return (
+      <span style={ { paddingBottom: '2px' } }>{ addonString }</span>
```
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.tsx]: i hate this padding here, but it's needed to get default text lined up with the text in the input. If you render an icon instead it seems to be fine, so I only put the styles on this span.

Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.tsx]: image

**Comment by micahgodbolt** [comment]
Looks like there's a bit of work to do to support the "underlined" input

**Comment by micahgodbolt** [comment]
Fixed underlined version. Put icon back as absolute to not break multiline resize handle.

**Comment by mikewheaton** [comment]
Are we sure we want to support this 'addons' functionality? We generally try to steer clear of anything that's not in our products.

File: packages/office-ui-fabric-react/src/components/TextField/TextField.scss [hunk 2]
```diff
@@ -106,13 +98,42 @@
...
     background-color: $ms-color-neutralLighter;
     border-color: $ms-color-neutralLighter;
     pointer-events: none;
     cursor: default;
   }
+}
+
+.fieldAddon {
+  background: #F4F4F4;
+  color: #666666;
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.scss]: no theme colors?

File: packages/office-ui-fabric-react/src/components/TextField/TextField.tsx [hunk 3]
```diff
@@ -225,6 +235,13 @@ export class TextField extends BaseComponent<ITextFieldProps, ITextFieldState> i
     }
   }
 
+  private _onRenderAddon(props): JSX.Element {
```
Comment (cschleiden) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.tsx]: type?

File: packages/office-ui-fabric-react/src/components/TextField/TextField.scss (hunk 2 above)
Comment (micahgodbolt) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.scss]: Hah...thanks for  catching that. Grabbed hex values from the designer, forgot to convert them.

**Pull request merged**